### PR TITLE
mgr/dashboard: CephFS snapshot management UI

### PR DESCRIPTION
**Pull request by Devp00l**
CephFS snapshots can now be created on a directory basis. Multiple
snapshots can be deleted at once.

Fixes: https://tracker.ceph.com/issues/41791

Signed-off-by: Volker Theile <vtheile@suse.com>
Signed-off-by: Stephan Müller <smueller@suse.com>

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by votdev** [comment]
```
> ceph-dashboard@0.0.0 i18n:extract /home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/dashboard/frontend
> ng xi18n --i18n-format xlf --i18n-locale en-US --output-path locale --progress=false && ngx-extractor -i 'src/**/*.ts' -f xlf -o src/locale/messages.xlf -l en-US

The following source translations in 'messages.xlf' need to be fixed, please check the I18N suggestions in 'HACKING.rst':
```

**Comment by votdev** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-directories/cephfs-directories.component.html [hunk 1]
```diff
@@ -25,9 +25,17 @@
...
 
         <legend i18n>Snapshots</legend>
-        <cd-table [sorts]="snapshot.sortProperties"
-                  [data]="selectedDir.snapshots"
-                  [columns]="snapshot.columns">
+        <cd-table [data]="selectedDir.snapshots"
+                  [columns]="snapshot.columns"
+                  identifier="name"
+                  forceIdentifier="true"
+                  selectionType="multi"
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-directories/cephfs-directories.component.html]: Bildschirmfoto vom 2019-11-08 12-43-55
```suggestion
                  selectionType="multiClick"
```

**Comment by Devp00l** [comment]
Addressed all comments and rebased the branch :)

**Pull request merged**